hoodie-client/src/main/java/com/uber/hoodie/config/HoodieCompactionConfig.java

```diff
@@ -104,6 +105,7 @@
   private static final String DEFAULT_CLEANER_COMMITS_RETAINED = "24";
   private static final String DEFAULT_MAX_COMMITS_TO_KEEP = String.valueOf(128);
   private static final String DEFAULT_MIN_COMMITS_TO_KEEP = String.valueOf(96);
+  private static final String DEFAULT_COMMITS_ARCHIVAL_BATCH_SIZE = String.valueOf(10);
```

Comment:
Is 10 a good default for both kafka/schemaless tables ? For cloud based stores, this number will also influence the number of archive files lying around permanently.

Ideally, if we have a SpillableList like implementation and use it here, we do need to batch and worry about OOM.

Reply:
This config is just driving how many commits to read and batch in a single outputstream.append(..) call, basically what would be the size of a single log block. This config does not influence the number of archive files lying around.

Reply:
On closer look, the outputstream is not closed per iterations. Should be fine.

Reply:
ok